Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -31,7 +31,10 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
         let oid = "5F07445CFBBBBBBBBBFAAAAA"
 
         // Success case
-        expect(try BSONObjectID(fromExtJSON: ["$oid": JSON.string(oid)], keyPath: [])).to(equal(try BSONObjectID(oid)))
+        let bson = try BSONObjectID(fromExtJSON: ["$oid": JSON.string(oid)], keyPath: [])
```

Comment:
it looks like we are missing tests for some of the types that got the new methods implemented on them. `Date`, `Bool`, maybe others

Reply:
We discussed this and figured that given that her next ticket is the corpus tests, it made sense to just relegate most of the testing to that. What do you think?

Reply:
Yeah, @patrickfreed  and I discussed not testing extensively for this ticket since the corpus tests will catch test all of this functionality and catch any issues

Reply:
I did add a few Date ones though

Reply:
that's fine with me; I was mainly thinking it would have been useful for catching the Date issue, but I agree it is probably redundant to be testing both.